src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/command/flow/ingress/of/IngressFlowModFactory.java

```diff
@@ -344,15 +344,17 @@ public OFFlowMod makeSingleVlanFlowLoopMessage() {
         RoutingMetadata metadata = RoutingMetadata.builder()
                 .outerVlanId(endpoint.getOuterVlanId())
                 .build(switchFeatures);
+        Match.Builder matchBuilder = of.buildMatch()
+                .setExact(MatchField.IN_PORT, OFPort.of(endpoint.getPortNumber()))
+                .setMasked(MatchField.METADATA,
+                OFMetadata.of(metadata.getValue()), OFMetadata.of(metadata.getMask()));
+        if (endpoint.getInnerVlanId() != 0) {
```

Comment:
This is the method for single VLAN flow, how it can have inner VLAN defined?

PS Please use `FlowEndpoint.isVlanIdSet()` method to check definement of vlan id.

Reply:
Fixed.